shell/platform/android/io/flutter/embedding/android/FlutterSplashView.java

```diff
@@ -0,0 +1,276 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.FrameLayout;
+
+import io.flutter.Log;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.renderer.OnFirstFrameRenderedListener;
+
+/**
+ * {@code View} that displays a {@link SplashScreen} until a given {@link FlutterView}
+ * renders its first frame.
+ */
+public class FlutterSplashView extends FrameLayout {
```

Comment:
I'll actually make this package private because it's just for Flutter. At that point I'm not aware of any reason to mark it final.

Reply:
SG! The same concerns about it getting potentially subclassed and the subclass breaking in weird ways apply, but it's a much smaller surface area now. If it's package private it could only be subclassed and then strangely broken within the namespace. It's still a bit of a risk, but a much smaller one.